### PR TITLE
 EventsSystem manager for Pointers

### PR DESCRIPTION
**Pull request by gonpombo8**
## What ?
Add EventsSystem callbacks =>  `onPointerDown`, `onPointerUp` & `onClick`
Remove EventsSystem callbacks => `removeOnPointerDown`, `removeOnPointerUp` & `removeOnClick`
- [x] Add helpers to manage PointerEvents
- [x] Fix InputAction.IA_ANY action. (It wasn't triggering anything)

## Why ?

Add callback style for onClick & onPointerDown fns to be more readable.
```ts
EventsSystem.onPointerDown(
  entity,
  function () {
    createCube(1 + Math.random() * 8, Math.random() * 8, 1 + Math.random() * 8)
    EventsSystem.removeOnPointerDown(entity)
  },
  {
    button: InputAction.IA_PRIMARY,
    hoverText: 'CASLA - BOEDO'
  }
)

EventsSystem.onClick(entity, fn, { button, hoverText })
EventsSystem.onPointerDown(entity, fn)
EventsSystem.onPointerUp(entity, fn)
EventsSystem.removeOnPointerUp(entity)
EventsSystem.removeOnPointerDown(entity)
EventsSystem.removeOnPointerUp(entity)

```

### CODE REVIEW

File: packages/@dcl/ecs/src/systems/events.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  ) => void | Promise<void>
+
+  export type Opts = {
+    button: InputAction
+    hoverText?: string
+  }
+  type EventMapType = Map<PointerEventType, { cb: Callback; opts: Opts }>
+
+  const defaultOpts: Opts = {
+    button: InputAction.IA_ANY,
```
Comment (gonpombo8) [inline packages/@dcl/ecs/src/systems/events.ts:20]: TODO: fix this. It's not working

Comment (gonpombo8) [reply packages/@dcl/ecs/src/systems/events.ts:20]: ✅

File: packages/@dcl/ecs/src/systems/events.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+          hoverText: opts.hoverText
+        }
+      })
+    }
+  }
+
+  function removeHoverFeedback(
+    entity: Entity,
+    type: PointerEventType,
+    button: Opts['button']
```
Comment (leanmendoza) [inline packages/@dcl/ecs/src/systems/events.ts:57]:
```suggestion
    button: InputAction
```

File: packages/@dcl/ecs/src/systems/events.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    event: PBPointerEventsResult_PointerCommand
+  ) => void | Promise<void>
+
+  export type Opts = {
+    button: InputAction
+    hoverText?: string
+  }
+  type EventMapType = Map<PointerEventType, { cb: Callback; opts: Opts }>
+
+  const defaultOpts: Opts = {
```
Comment (leanmendoza) [inline packages/@dcl/ecs/src/systems/events.ts:19]:
```suggestion
  const getDefaultOptions: Opts = () => ({
    button: InputAction.IA_ANY,
    hoverText: ''
  })
```
I think it's not the case right now, but if we could edit hoverFeedback or the button, it'll modify all the events with reference to the default.

File: packages/@dcl/ecs/src/systems/events.ts [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    removeEvent(entity, PointerEventType.PET_DOWN)
+  }
+
+  /**
+   * Execute callback when the user pressed the button declared while pointing at the entity
+   * @param entity Entity to attach the callback
+   * @param cb Function to execute when onPointerDown fires
+   * @param opts Opts to trigger Feedback and Button
+   */
+  export function onClick(entity: Entity, cb: Callback, opts = defaultOpts) {
```
Comment (leanmendoza) [inline packages/@dcl/ecs/src/systems/events.ts:101]:
```suggestion
  export function onClick(entity: Entity, cb: Callback, opts = getDefaultOptions()) {
```

File: packages/@dcl/ecs/src/systems/events.ts [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  /**
+   * Execute callback when the user releases the button declared while pointing at the entity
+   * @param entity Entity to attach the callback
+   * @param cb Function to execute when click fires
+   * @param opts Opts to trigger Feedback and Button
+   */
+  export function onPointerDown(
+    entity: Entity,
+    cb: Callback,
+    opts = defaultOpts
```
Comment (leanmendoza) [inline packages/@dcl/ecs/src/systems/events.ts:119]:
```suggestion
    opts = getDefaultOptions()
```

File: packages/@dcl/ecs/src/systems/events.ts [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+import { Entity, IEngine, IInput } from '../engine'
+
+declare const engine: IEngine
+
+export namespace EventsSystem {
+  export type Callback = (
+    event: PBPointerEventsResult_PointerCommand
+  ) => void | Promise<void>
+
+  export type Opts = {
```
Comment (leanmendoza) [inline packages/@dcl/ecs/src/systems/events.ts:13]:
```suggestion
  export type Options = {
```
A more verbose?

**Review by leanmendoza — APPROVED** [review]
Great job 😄 I left some comments and optional changes. When the time to name `Input` namespace comes, we're going to have to take this in the same alignment.

**Review by leanmendoza — APPROVED** [review]

**Pull request merged**